### PR TITLE
Create boefje variant

### LINKED ISSUE #3403: Creating duplicate boefjes
_This ticket is part of epic #2736_ 

------------

### Detailed description
This is the implementation ticket of design ticket #2870 

### Feature benefit/User story
As a KAT user, I want to be able to duplicate boefjes, so I can easily create a new Boefje based on the settings of another Boefje.

### Specifications
The implementation should include…

- [ ] "Add variant" button on Boefje detail page
- [ ] "Boefje variant setup" page with a form with pre-filled input fields
- [ ] Add variant to section on Boefje detail page
- [ ] Set permission to admin only

### Screenshots

_Include screenshots of the proposed design changes here._

### Figma link
[Approved | Boefjes | Duplicate a Boefje](https://www.figma.com/design/D48Pf9lLpxET1jp0Wsjzw5/%E2%9C%85-KAT-%7C-Approved-%7C-Boefjes?node-id=2301-14554&t=OkC1Ba5bbLcxjHzG-1)

[For Review, not final! ](https://www.figma.com/design/wTD4hfTbZMp0o2G6VmUepU/%E2%9C%8D%F0%9F%8F%BC-KAT-%7C-For-Review-%7C-Boefjes?node-id=2605-40514)

## Implementation

_This part should only be filled in by the developers._

### Possible solution

_Outline your proposed solution for implementing the feature. You can include any specific ideas, designs, or functionalities here._

### Alternatives considered

_Describe any alternative approaches or solutions you've considered, and why you believe the proposed solution is superior._

### PR DESCRIPTION
**Pull request by madelondohmen**
**NOTE:** This PR is a follow-up of PR #3400

------------------------

### Changes
This PR makes it possible to add a new Boefje, based on another Boefje.

Features added:
- "Add variant" button on Boefje detail page
- "Setup Boefje variant" form page that is pre-filled
   - Changes in the helper texts
   - Changed "Input object type" single select to multi select
   - This form page redirects to original Boefje detail page
- Boefje detail page
   - Variant should be added in the table
   - Confirmation should be visible
- Backend: add filter (oci_image) to get_plugins query, so we can collect all variants of one specific oci_image

### Issue link
Closes #3403 

### Demo
afbeelding
afbeelding

### QA notes
- Go to "Nmap TCP" boefje
- Check the "Variants" table. There should be one item (the current Boefje)
- Click "Add variant"
- Check if all the input fields are filled in (except for "name" and "description")
- Type a different name, make some small changes to the other input fields
- Click "Create variant"
- You should return to the "Nmap TCP" boefje
   - In the variants table, a new variant should be visible
   - There should be a confirmation message that the new variant has been created successfully
- Click on the name of the new variant and check the Boefje details page

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [x] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/katalogus/templates/plugin_container_image.html [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                <a class="button ghost"
+                   href="{% url "boefje_variant_setup" plugin_id=plugin.id organization_code=organization.code %}"><span aria-hidden="true" class="icon ti-plus"></span>{% translate "Add variant" %}</a>
+            </div>
+        {% endif %}
+    </div>
+    {% if variants %}
+        {% if new_variant %}
+            <p class="confirmation"
+               role="group"
+               aria-label="{% translate "new variant confirmation" %}">
```
Comment (ppvg) [inline rocky/katalogus/templates/plugin_container_image.html:30]: There's been some discussion on whether the `role="group"` is helpful or harmful for notifications. We may be removing this recommendation from the Manon docs. I would advise removing it, as it has the potential to _improve_ the accessibility of something that is already accessible, but also has the potential to make it _less_ accessible.

```suggestion
            <p class="confirmation">
```

If you do want to keep it, consider changing the label to just "confirmation". The role and label are meant as a text alternative for the visual style of the notification. That is: "visually distinct green block" == "notification".

The same applies to other notifications, like the "no variants explanation" (either remove role and label or change label to just "explanation").

**Review by ammar92 — COMMENTED** [review]
Just a few remarks. It's quite a big PR so I might have missed a few spots, but it looks good in general. There should be some careful testing

**Comment by underdarknl** [comment]
https://github.com/minvws/nl-kat-coordination/pull/3400 is now merged.

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Works now! Can create variants on the nmap boefje. They run and get picked up by the normalizer. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Review by underdarknl — APPROVED** [review]

File: rocky/rocky/views/tasks.py [hunk 2]
```diff
@@ -65,6 +66,24 @@ class NormalizersTaskListView(TaskListView):
...
+        ids = [
+            task.data.raw_data.boefje_meta.boefje.id
+            for task in task_list
+            if task.data.raw_data.boefje_meta.boefje.id != "manual"
+        ]
+        plugins = get_katalogus(self.organization.code).get_plugins(ids=ids)
+        plugin_dict = {p.id: p.name for p in plugins}
+
+        for task in task_list:
+            task.data.raw_data.boefje_meta.boefje.name = plugin_dict.get(task.data.raw_data.boefje_meta.boefje.id)
```
Comment (underdarknl) [inline rocky/rocky/views/tasks.py:83]:
```suggestion
            task.data.raw_data.boefje_meta.boefje.name = plugin_dict[task.data.raw_data.boefje_meta.boefje.id]
```

**Review by ammar92 — APPROVED** [review]
Looks good to me. Tiny merge conflict to be fixed, but should be ready to merge soon

**Review by ammar92 — APPROVED** [review]

**Pull request merged**